zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -1150,4 +1140,33 @@ public void setFirstPacket(boolean value) {
         needOpPacket = value;
     }
 
+    void closeSocket() {
+        if (sock != null && !sock.isClosed() && sockBeingClosed.compareAndSet(false, true)) {
+            LOG.error("Unexpected exception causing shutdown while sock is still open");
```

Comment:
if I am right, when we are here, we already logged out the exception, if we had any.
But we can also get here from the `shutdown()` function, when we close the socket just without  catching any exceptions.

What about instead of `LOG.error("Unexpected exception...` just print out a `LOG.info` about closing the socket, and also indicating in the logs if we will do it sync or async?

Reply:
You're right. The message is confusing. Changed.